### ISSUE TITLE
Make modelled data available to 3rd parties

### ISSUE THREAD

**Issue by dotloadmovie** [SocialFinanceDigitalLabs/sf-fons-platform#49]
**Budget/Funding Update from Tom R**
24th Jan 2024:  Unknown

**Need to consider how this might work for EoE.  @MichaelHanksSF** 

originally was API.. but now it is ability to dump files into their Azure bucket.

**Business Case:**
-
- It is critical and time-sensitive that we provide the capability for this from 'Day 1' of any LA being operational  on the Platform
•	Fons used to collect data from many organisations and prepare it for analysis, and then data needs to be handed to 3rd party for further processing
•	Data is high risk (personal, damaging if released, large scale, describing vulnerable individuals, collected by an organisation with power over them)
•	3rd party requires sufficient metadata to be able to display in their own products information on how any figures they are showing have been arrived at (e.g. what dataset uploads is this based on, how have they been manipulated to arrive at the data we're handing over)
•	3rd party may often be one with questionable interest/capability as regards information security, but also one who we lack the power to replace or to di-rect improvement in. Therefore we may restrict the solutions we're willing to support to those which it is easiest for the 3rd party to implement securely.

**Problem Statement:**
-
- We need to be able to make data that is uploaded, stored and processed in an instance of the Platform available to the Commissioning Alliance so that they can create analysis of it and distribute it to decision makers in 32 boroughs using their own Power BI and Active Directory setup
- We need to ensure that we make this as low-risk an option for them in regard to security/data protection as possible but this is no longer re APIs it is and must be via their Azure bucket

**Data Sets In Scope:**
-
- SSSDA903  for now- make it replicable for new sets of data
- CIN
- Annex A

**Use Case(s):**
-

**1. CA:**

As the Commissioning Alliance's Analyst I need a way to ...
- receive processed output data from the Data Platform securely and appropriately 
- continue to  provide analysis and distribute this analysis to the 33 London Borough 'decision makers'

Unknowns:
•	Cadence of data updates, and predictability of their timing - we don't yet know how often there will be a change to the data on our systems because this is a new service. Remember uploading new data may be a result of LAs realising the old data contained errors so is not necessarily just once a year.
•	We don't know how quickly LAs will expect to see the impact of their upload of new data into the system, or how important either speed or predictability of data refresh will be to them. Seems fair to assume that 'by 9am next working day' will do for now but this one feels like a relatively risky assumption.
•	What notifications if any Commissioning Alliance need from us (e.g. when updated data is available)
•	[@Dave Tickle](https://socialfinance.slack.com/team/U423D9AVC) this might be a useful place for you to call out anything you need more clarity on

Not part of the requirement, but possibly useful to know:
•	In other instances, the BI tools being used by the partner may change (e.g. NHS NWL use Tableau)
•	We're also creating our own front-end for analysis of data held in Fons, which will be used initially just for demand modelling.

**3. Duties:**

- To continue to analyse Pan Region Aggregated (PanAgg) data and fulfil my duties
- To continue to produce reports for stakeholders and the Local Authority as expected
More to come from Tom

**5. IG Considerations:**

Does the current IG cover this? - [ ] Yes      - [ ] No

Other IG notes and/or actions:

**6. Technical Proposal:**
-
•	We propose the use of Azure Block Storage to hold the data for PowerBI, under the control of CA or their technical partner
•	SF will send data there directly from the pipeline (as opposed to putting it on-to our own S3 storage)
•	SF will assist the creation of this Azure storage location if needed, by providing documentation about what should be created, where and how
•	PowerBI will be able to use this Azure Block Storage location natively to retrieve the files - an option for this is available in the menuing inside PowerBI

Quick

**Estimated cost to deliver:**

Development time: 3 Sprints (6 weeks), 2 Developers (agile deployment of different Developers as skills are needed)

END

Historic info: 
Option 1:  Get CA to do the work
Option 2:  Synthetic data
Feels like work effort for both/either options is same so solution will be ready to hand off to others regardless.
a) Or can we stand up our own Power BI.
This currently relies on Simpsons.
and issues with CA, and know Herts not continuing with them long term.
Issue: We don't have a copy of Ofsted file and no one is allowed to give it to us. (PanAgg and Ofsted provision data = Dynamic Sufficiency)
Ofsted agreement is that LAs get this on monthly basis but LAs are not allowed to share it outside.
Conclusion: We cannot do our own Power BI 
b) Herts set up to do the work. A lot of dependencies and does not necessarily run any faster. Known challenges with LAs and IT teams.  TR will still warm Herts to see if they could take on the CA role to run in parallel.  Will also ask to see the wording of the Ofsted agreement re: the Ofsted file. (see if we CAN receive it in some shape or form).

we need to start thinking about API on the egress layer. Initially this is probably going to be as simple as exposing a file so that an external process can curl it. I think this presents a handful of (hopefully minor) issues:

How do we gate access (can the normal SSO handle this)
What do we expose (REST, WEBDAV, another?)
Do we expose on a per-LA basis?

**Comment by MagicMiranda** [comment]
https://socfinuk.atlassian.net/browse/SFDPCM-5

**Comment by MichaelHanksSF** [comment]
Push data from 'hub' instance to EoE external Azure storage
Herts Credentials provided and Sf PUSH to them

Push data from 'hub' instance to London Councils external Azure storage
LC Credentials provided and Sf PUSH to them

**Comment by dotloadmovie** [comment]
[fiona.judd@hertfordshire.gov.uk](mailto:fiona.judd@hertfordshire.gov.uk)

**Comment by MichaelHanksSF** [comment]
- Test PUSH from FONS instance to Conosco-created Azure blob storage SPRINT 1
- Tee up the arrangement with Herts for PUSH to their storage SPRINT 1
- EoE generate service user to allow PUSH from platform; contact at EoE is Fiona Judd ICT Project Manager SPRINT 2/3

**Comment by MichaelHanksSF** [comment]
- Michael to get Rashid to share right details for sharing
- Tee up the arrangement with London Councils for PUSH to their storage SPRINT 1
- London Councils generate service user to allow PUSH from platform; contact at LC is ??? SPRINT 2/3

**Comment by MichaelHanksSF** [comment]
Dependent on #68
